postreise/plot/plot_carbon_bar.py

```diff
@@ -1,51 +1,77 @@
-        as returned by :func:`postreise.analyze.carbon.summarize_carbon_by_bus`.
-    :param str scenario_name_1: name of first scenario, for labeling plot.
-    :param str scenario_name_2: name of second scenario, for labeling plot.
+def plot_carbon_bar(*args, labels=None, labels_size=15):
+    """Make bar chart of carbon emissions by fuel type for n scenarios.
+
+    :param powersimdata.scenario.scenario.Scenario args: scenario instances.
+    :param list/tuple/set labels: labels on plot. Default to scenario id.
+    :param int labels_size: size of labels.
+    :raises ValueError:
+        if ``args`` are not scenario instances.
+        if ``labels`` has a different length than the number of passed scenarios.
+    :raises TypeError:
+        if ``labels`` is not an iterable.
+        if ``labels_size`` is not an int.
     """
-    plt.rcParams.update({"font.size": 22})
-    objects = (scenario_name_1, scenario_name_2)
-    y_pos = np.arange(len(objects))
-
-    carbon_val = [
-        sum(carbon_by_bus_1["coal"].values()),
-        sum(carbon_by_bus_2["coal"].values()),
-    ]
-    _ = plt.figure(figsize=(7, 2))
-    plt.barh(y_pos, carbon_val, align="center", alpha=0.25, color=["black"])
-    plt.yticks(y_pos, objects)
-    plt.xlabel("Tons")
-    plt.title("Coal: CO$_2$  Emissions", y=1.04)
-    plt.show()
+    if not all([isinstance(a, Scenario) for a in args]):
+        raise ValueError("all inputs must be Scenario objects")
+    if labels is not None and not isinstance(labels, (list, tuple, set)):
+        raise TypeError("labels must be a list/tuple/set")
+    if labels is not None and len(args) != len(labels):
+        raise ValueError("labels must have same length as number of scenarios")
+    if not isinstance(labels_size, int):
+        raise TypeError("labels_size must be a int")
+
+    labels = tuple([s.info["id"] for s in args]) if labels is None else tuple(labels)
 
-    ng_val = [sum(carbon_by_bus_1["ng"].values()), sum(carbon_by_bus_2["ng"].values())]
+    carbon_val = {"coal": [], "ng": []}
+    for i, s in enumerate(args):
+        grid = s.state.get_grid()
+        carbon_by_bus = summarize_emissions_by_bus(generate_emissions_stats(s), grid)
+        carbon_val["coal"].append(sum(carbon_by_bus["coal"].values()))
+        carbon_val["ng"].append(sum(carbon_by_bus["ng"].values()))
 
-    _ = plt.figure(figsize=(7, 2))
-    plt.barh(y_pos, ng_val, align="center", alpha=0.25, color=["purple"])
-    plt.yticks(y_pos, objects)
-    plt.xlabel("Tons")
-    plt.title("Natural Gas: CO$_2$ Emissions", y=1.04)
+    fig, (ax1, ax2) = plt.subplots(1, 2, sharey=True, figsize=(14, len(labels)))
+    y_pos = np.arange(len(labels))
+
+    for a, f, c, t in zip(
+        [ax1, ax2],
+        ["coal", "ng"],
+        ["black", "purple"],
+        ["Coal: CO$_2$  Emissions", "Natural Gas: CO$_2$ Emissions"],
+    ):
+        a.barh(y_pos, carbon_val[f], align="center", alpha=0.25, color=c)
+        a.set_xlabel("Tons", fontsize=15)
```

Comment:
Should we use `labels_size` here as well?

Reply:
Well, I am not sure. The labels are the ticks on the y-axis, this one (Tons) is the title on the x-axis usually the title of the axis is larger than the labels of the ticks of the axis.

Reply:
They are currently in the same size 15, no? Or we can use the `labels_size` as a reference and put down relative values for other font sizes if we would like to keep the relationship, something like `fontsize=labels_size - 2`.

Reply:
Done